src/middlewares/authentication/base/authentication.middleware.ts

```diff
@@ -0,0 +1,34 @@
+import admin from 'firebase-admin';
+import { Request, Response, NextFunction } from 'express';
+
+import * as ERROR_MSG from '../../../constants/authErrorMessages';
+
+const AuthMiddleware = (permittedRoles: Array<String>) => (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const { token } = req.body;
+
+  admin
```

Comment:
let's rip this out into like auth.service.ts which exports a method called verifyRole or something - it'll make testing a lot easier.

Reply:
You mean the whole base AuthMiddleware? And what will verifyRole take as parameters? Sorry I'm just a little confused

Reply:
just wrapping the admin.VerifyIdToken stuff into a method that we own to decouple our controller from firebase to make testing easier

Reply:
Oh so move the entire admin.verifyIdToken block to another file gotcha

Reply:
you can only use the await keyword in an async function